backend/app/modules/user/patron_mode/service/events_based.py

```diff
@@ -26,3 +29,37 @@ def get_patrons_rewards(self, context: Context) -> int:
         return database.budgets.get_sum_by_users_addresses_and_epoch(
             patrons, epoch_details.epoch_num
         )
+
+    def get_patron_donations(
+        self, context: Context, user_address: str, from_timestamp: Timestamp, limit: int
+    ) -> list[PatronDonationItem]:
+        last_finalized_snapshot = context.snapshots_state.last_finalized_snapshot_num
+
+        epochs = epoch_details.get_epochs_details(
+            last_finalized_snapshot - limit, last_finalized_snapshot
+        )
+        epochs.reverse()
+
+        events = []
+
+        for epoch in epochs:
```

Comment:
(not blocking; nitpicking)
I know it's copy-pasted from the lagacy but since it looks like ```arrow anti-pattern```, maybe we should consider splitting the logic into several submethods / or try to simplify the ifs?

Reply:
ok, cleaned up